scrapy/settings/default_settings.py

```diff
@@ -87,7 +87,7 @@
 DOWNLOADER_MIDDLEWARES_BASE = {
     # Engine side
     'scrapy.downloadermiddlewares.robotstxt.RobotsTxtMiddleware': 100,
-    'scrapy.downloadermiddlewares.httpauth.HttpAuthMiddleware': 300,
+    'scrapy.downloadermiddlewares.auth.AuthMiddleware': 300,
```

Comment:
This is my only concern, existent projects changing priority of httpauth middleware will load the middleware twice (or once if disabled). 

I recall @curita implemented a solution for this case that seems to be at `scrapy/utils/deprecate.py` under `DEPRECATION_RULES`